### PR TITLE
Pick changes from 7.7.7

### PR DESCRIPTION
**Pull request by tsuoanttila**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/framework/8577)

### CODE REVIEW

**Comment by pleku** [comment]
Reviewed 1 of 6 files at r1, 4 of 24 files at r3, 3 of 4 files at r4, 2 of 2 files at r5.
Review status: 10 of 39 files reviewed at latest revision, 1 unresolved discussion.

---

*[compatibility-client/src/main/java/com/vaadin/v7/client/widgets/Grid.java, line 8786 at r5](https://reviewable.io:443/reviews/vaadin/framework/8577#-Kd5Lb-qHVdiw3DbeZdi:-Kd5Lb-r17J_CykHZql7:bs3gf4d) ([raw file](https://github.com/vaadin/framework/blob/ae1308bbc19978932fa92e28c48f9d484e2c5618/compatibility-client/src/main/java/com/vaadin/v7/client/widgets/Grid.java#L8786)):*
> ```Java
>         for (int row : details) {
>             setDetailsVisible(row, false);
>         }
> ```

This is now done twice!!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8577)*

**Comment by pleku** [comment]
Reviewed 5 of 6 files at r1, 8 of 8 files at r2, 17 of 24 files at r3, 1 of 1 files at r6, 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/Grid.java, line 2557 at r7](https://reviewable.io:443/reviews/vaadin/framework/8577#-KdViYwbAklqBrdFKknj:-KdViYwbAklqBrdFKknk:b-n22odv) ([raw file](https://github.com/vaadin/framework/blob/0afd53934e49a568cb804b57e8422e76899253cc/compatibility-server/src/main/java/com/vaadin/v7/ui/Grid.java#L2557)):*
>                 for (CELLTYPE cell : cellGroups.values()) {
>                     cell.detach();
>                 }

This is now done twice

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8577)*

**Comment by tsuoanttila** [comment]
*[compatibility-server/src/main/java/com/vaadin/v7/ui/Grid.java, line 2557 at r7](https://reviewable.io:443/reviews/vaadin/framework/8577#-KdViYwbAklqBrdFKknj:-KdVksF-WBAEZmDasxFZ:b-ecepzg) ([raw file](https://github.com/vaadin/framework/blob/0afd53934e49a568cb804b57e8422e76899253cc/compatibility-server/src/main/java/com/vaadin/v7/ui/Grid.java#L2557)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

>                 for (CELLTYPE cell : cellGroups.values()) {
>                     cell.detach();
>                 }

This is now done twice
</blockquote></details>

cells vs.cellGroups. This is the exact change that was in the original patch.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8577)*
